sqllogictest/src/runner.rs

```diff
@@ -1096,6 +1113,138 @@ mod tests {
         .run()
     }
 
+    #[test]
+    fn test_statement_query_output() {
+        TestCase {
+            // input has no query results
+            input: "statement ok\n\
+                    create table foo;",
+
+            // Model a run that produced a 3,4 as output
+            record_output: query_output(&[&["3", "4"]]),
+
+            expected: Some(
+                "statement ok\n\
+                 create table foo;",
+            ),
+        }
+        .run()
+    }
+
+    #[test]
+    fn test_query_statement_output() {
+        TestCase {
+            // input has no query results
+            input: "query III\n\
+                    select * from foo;\n\
+                    ----",
+
+            // Model a run that produced a statement output
+            record_output: statement_output(3),
+
+            expected: Some(
+                "statement ok\n\
+                 select * from foo;",
+            ),
+        }
+        .run()
+    }
+
+    #[test]
+    fn test_statement_output() {
+        TestCase {
+            // statement that has no output
+            input: "statement ok\n\
+                    insert into foo values(2);",
+
+            // Model a run that produced a statement output
+            record_output: statement_output(3),
+
+            // TODO: expect the output includes 3?
```

Comment:
I found it surprising that the statement output did not include the `3`. If you agree this is not expected I suggest we merge these tests and then fix the bug (and update the tests) as a follow on PR.

Reply:
I think it's expected behavior. The logic is: if the record is `statement count <n>`, `n` will be updated to real count. If the record is `statement ok` (which means we don't care the number of affected rows), it won't be updated. 

Do you think that makes sense?

Reply:
Also `statement count <n>` is a syntax not included in sqlite/duckdb's dialet. This approach can avoid messing up their test cases.

Reply:
Makes sense -- I have updated the comments